site/src/sphinx/advanced-zookeeper.rst

```diff
@@ -60,19 +60,41 @@ For more information, please refer to the API documentation of the `com.linecorp
 Automatic service registration
 ------------------------------
 
-Use `ZooKeeperUpdatingListener`_ to register your server to a ZooKeeper cluster:
+Use `ZooKeeperUpdatingListenerBuilder`_ to register your server to a ZooKeeper cluster:
 
 .. code-block:: java
 
     import com.linecorp.armeria.server.ServerListener;
-    import com.linecorp.armeria.server.zookeeper.ZooKeeperUpdatingListener;
+    import com.linecorp.armeria.server.zookeeper.ZooKeeperUpdatingListenerBuilder;
 
     // This constructor will use server's default host name, port and weight.
-    // Use the other constructors to override the defaults.
-    ServerListener listener = new ZooKeeperUpdatingListener(
-            /* zkConnectionStr */ "myZooKeeperHost:2181",
-            /* zNode           */ "/myProductionEndpoints",
-            /* sessionTimeout  */ 10000);
+    // Use `nodeValueCodec` method to override the defaults.
+    ServerListener listener =
+            // Zookeeper Connection String and Zookeeper Node to use
+            new ZooKeeperUpdatingListenerBuilder("myZooKeeperHost:2181", "/myProductionEndpoints")
+            // Session Timeout
+            .sessionTimeout(10000)
+            .build();
+    server.addListener(listener);
+    server.start();
+    ...
+
+You can use `CuratorFramework`_ instance instead of Zookeeper Connection String. In this case, you can use the
+`CuratorFramework`_ instance for the other purposes.
+
+.. code-block:: java
+
+    import com.linecorp.armeria.server.ServerListener;
+    import com.linecorp.armeria.server.zookeeper.ZooKeeperUpdatingListenerBuilder;
+    import org.apache.curator.framework.CuratorFramework;
+
+    CuratorFramework client = ...
+    ServerListener listener =
+            // CuratorFramework instance
+            new ZooKeeperUpdatingListenerBuilder(client, "/myProductionEndpoints")
+            // Server's default host name, port and weight
+            .nodeValueCodec(NodeValueCodec.DEFAULT)
+            .build();
```

Comment:
- Please fix indentation
- Probably OK to omit the call to `nodeValueCodec()`?
  - How about just using `ZooKeeperUpdatingListener.of(...)` instead of a builder here, and adding another section about how to use `nodeValueCodec()`?

Reply:
How about this?

- Change `NodeValueCodec.DEFAULT` into `CustomNodeValueCodec`.
- Add additional documentation on `ZooKeeperUpdatingListener.of(...)`.

Reply:
Could you explain why `NodeValueCodec.DEFAULT` has to be renamed to `CustomNodeValueCodec`?

Reply:
> Could you explain why `NodeValueCodec.DEFAULT` has to be renamed to `CustomNodeValueCodec`?

By doing so, we can provide an example of how to specify `NodeValueCodec` instance to use.

Reply:
Could you just leave this as it is then? Let me address it later.